Document the simdjson PECL

TysonAndre opened the pull request:
API documentation and other example snippets are available at https://github.com/crazyxman/simdjson_php

Depends on https://github.com/php/doc-base/pull/71 to build

cmb69 commented on reference/simdjson/simdjsonexception.xml:
```diff
@@ -0,0 +1,78 @@
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('exception.synopsis')/descendant::db:fieldsynopsis)"><xi:fallback/></xi:include>
+
+    
+    <classsynopsisinfo role="comment">&Methods;</classsynopsisinfo>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.simdjsonexception')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[not(@role='procedural')])"><xi:fallback/></xi:include>
+    
```
`SimdJsonException` doesn't have any methods (except for inherited ones), so:
```suggestion
```

cmb69 commented on reference/simdjson/simdjsonexception.xml:
```diff
@@ -0,0 +1,78 @@
+    <classsynopsisinfo role="comment">&Methods;</classsynopsisinfo>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.simdjsonexception')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[not(@role='procedural')])"><xi:fallback/></xi:include>
+    
+    <classsynopsisinfo role="comment">&InheritedMethods;</classsynopsisinfo>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.runtimeexception')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[not(@role='procedural')])"><xi:fallback/></xi:include>
+
```
`RuntimeException` inherits its methods from `Exception`, so:
```suggestion
    <classsynopsisinfo role="comment">&InheritedMethods;</classsynopsisinfo>
    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.exception')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[not(@role='procedural')])"><xi:fallback/></xi:include>

```

cmb69 commented on reference/simdjson/versions.xml:
```diff
@@ -0,0 +1,57 @@
+ <!-- Classes and Methods -->
+
+ <function name='simdjsonexception' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::__clone' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::__construct' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::__wakeup' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::getmessage' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::getcode' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::getfile' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::getline' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::gettrace' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::getprevious' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::gettraceasstring' from='PECL simdjson &gt;= 2.1.0'/>
+ <function name='simdjsonexception::__tostring' from='PECL simdjson &gt;= 2.1.0'/>
```
I don't think this is necessary, since these are inherited.

cmb69 commented on reference/simdjson/versions.xml:
```diff
@@ -0,0 +1,57 @@
+ <function name='simdjsonexception::__tostring' from='PECL simdjson &gt;= 2.1.0'/>
+
+ <function name='simdjsonvalueerror' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::__clone' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::__construct' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::__wakeup' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::getmessage' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::getcode' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::getfile' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::getline' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::gettrace' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::getprevious' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::gettraceasstring' from='PECL simdjson &gt;= 3.0.0'/>
+ <function name='simdjsonvalueerror::__tostring' from='PECL simdjson &gt;= 3.0.0'/>
```
I don't think this is necessary, since these are inherited.

cmb69 reviewed (commented):
Thank you! This still needs some work, but is a great start!

cmb69 commented on reference/simdjson/setup.xml:
```diff
@@ -0,0 +1,43 @@
+   <link xlink:href="&url.pecl.package;simdjson">&url.pecl.package;simdjson</link>.
+  </para>
+  <para>
+   &pecl.windows.download.avail;
```
Well, not really, but just in case anybody asks, there are [**unofficial** builds of simdjson 4.0.0](https://github.com/cmb69/php-ftw/actions/runs/3289880012) available.

TysonAndre commented on reference/simdjson/constants.xml:
```diff
@@ -0,0 +1,358 @@
+    <listitem>
+     <simpara>
+      This parser can't support a document that big.
+      Thrown when parsing a JSON string that is over 4GiB long.
```
See https://github.com/simdjson/simdjson/blob/master/doc/basics.md#standard-compliance (using a fork of this C library to match the php json_decode results for numbers)

> - The specification states that an implementation may set limits on the size of texts that it accepts. The simdjson library limits single JSON documents to 4 GiB. It will refuse to parse a JSON document larger than 4294967295 bytes.

TysonAndre commented on reference/simdjson/functions/simdjson-decode.xml:
```diff
@@ -0,0 +1,302 @@
+    <programlisting role="php">
+<![CDATA[
+<?php
+// Encode some data with a maximum depth of 4 (array -> array -> array -> string)
```
These snippets are the same as https://www.php.net/json_decode , but I agree

cmb69 commented on reference/simdjson/book.xml:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<book xml:id="book.simdjson" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <title>Simdjson</title>
```
To satisfy the extension checker (`doc-base/scripts/qa/extensions.xml.php --check`):
```suggestion
 <?phpdoc extension-membership="pecl" ?>
 <title>Simdjson</title>
```

cmb69 reviewed (commented):
I've pushed a commit which removes trailing whitespace (to satisfy the whitespace checker). Otherwise this should be good now except for the `extension-membership`.

TysonAndre commented:
> The whitespace checker still fails, probably because not all commits are free of trailing whitespace. Either squash the commits, or just ignore that.

I'd forgotten about that - I had whitespace trimming turned off from previous work on editing existing doc files. I pulled your changes and fixed remaining trailing whitespace and squashed.

cmb69 commented:
Thank you!
